k-distribution/k-tutorial/2_intermediate/01_macros/README.md

```diff
@@ -4,6 +4,151 @@ copyright: Copyright (c) Runtime Verification, Inc. All Rights Reserved.
 
 # Lesson 2.1: Macros, Aliases, and Anywhere Rules
 
+The purpose of this lesson is to explain the behavior of the `macro`,
+`macro-rec`, `alias`, and `alias-rec` production attributes, as well as the
+`anywhere` rule attribute. These attributes control the meaning of how rules
+associated with them are applied.
+
+## Macros
+
+Thus far in the K tutorial, we have described three different types of rules:
+
+1. Top-level rewrite rules, which rewrite a configuration composed of cells to
+another configuration;
+2. Function rules, which define the behavior of a function written over
+arbitrary input and output types; and
+3. Simplification rules, which describe ways in which the symbolic execution
+engine ought to simplify terms containing symbolic values.
+
+This lesson introduces three more types of rules, the first of which are
+**macros**. A production is a macro if it has the `macro` attribute, and all
+rules whose top symbol on the left hand side is a macro are **macro rules**
+which define the behavior of the macro. Like function rules and simplification
+rules, macro rules do not participate in cell completion. However, unlike
+function rules and simplification rules, macro rules are applied **statically**
+before rewriting begins, and the macro symbol is expected to no longer appear
+in the initial configuration for rewriting once all macros in that
+configuration are rewritten.
+
+The rationale behind macros is they allow you to define one piece of syntax
+in terms of another piece of syntax without any runtime overhead associated
```

Comment:
Might be worth mentionng that this is a common practice in PLs called "desugaring"